crates/sui-types/src/messages_consensus.rs

```diff
@@ -34,12 +48,15 @@ pub struct ConsensusTransaction {
     pub kind: ConsensusTransactionKind,
 }
 
-#[derive(Serialize, Deserialize, Clone, Copy, Hash, PartialEq, Eq)]
+#[derive(Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
 pub enum ConsensusTransactionKey {
     Certificate(TransactionDigest),
     CheckpointSignature(AuthorityName, CheckpointSequenceNumber),
     EndOfPublish(AuthorityName),
     CapabilityNotification(AuthorityName, u64 /* generation */),
+    // Key must include both id and jwk, because honest validators could be given multiple jwks for
```

Comment:
Btw, I wonder if this could have been prevented, if we chose JwkId = Hash(JWK)? Is there reason we chose a schema that allows multiple JWKs with same id?

Reply:
If we were generating the data ourselves, we could do something like that. However, a malfunctioning oauth provider could actually return conflicting jwks. Similarly, the same JWK could be named by more than one ID, and a hash wouldn't handle that case correctly. The first case will probably never happen, but the second case seems at least conceivable to me and I'll sleep better knowing we handle it correctly.